### PR TITLE
Remove various defers from k8s_test.go

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description

Fixes https://github.com/projectcalico/libcalico-go/issues/716

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: lib/backend/k8s/k8s_test.go [hunk 1]
```diff
@@ -320,6 +320,41 @@ var _ = Describe("Test Syncer API for Kubernetes backend", func() {
...
+		// Clean up any k8s network policy left over by test.
+		nps := networkingv1.NetworkPolicyList{}
+		err := c.clientSet.NetworkingV1().RESTClient().
+			Get().
+			Resource("networkpolicies").
+			Timeout(10 * time.Second).
+			Do().Into(&nps)
+		Expect(err).NotTo(HaveOccurred())
+
+		for _, np := range nps.Items {
```
Comment (gunjan5) [inline lib/backend/k8s/k8s_test.go]: could we not just delete the namespace to get rid of all these things? I guess we can't delete the default namespace, but maybe if we create a namespace for every test (kinda like upstream k8s NetworkPolicy e2es) and just create the namespace in `BeforeEach` and delete it in `AfterEach`? Maybe it's too much work or we need to make sure things work in the default namespace, in which case this is fine.

File: lib/backend/k8s/k8s_test.go [hunk 2]
```diff
@@ -491,39 +503,16 @@ var _ = Describe("Test Syncer API for Kubernetes backend", func() {
...
-					Timeout(10 * time.Second).
-					Do().Into(&nps)
-				Expect(err).NotTo(HaveOccurred())
-				time.Sleep(1 * time.Second)
-			}
-			panic(fmt.Sprintf("Failed to clean up policies: %+v", nps))
-		})
-	}()
+		// Delete the object.
+		result := c.clientSet.NetworkingV1().RESTClient().
```
Comment (gunjan5) [inline lib/backend/k8s/k8s_test.go]: bit confused, why do we need this one? Shouldn't it be taken care of in `AfterEach`? I'm sure there's a reason, but im blanking out

File: lib/backend/k8s/k8s_test.go [hunk 3]
```diff
@@ -491,39 +503,16 @@ var _ = Describe("Test Syncer API for Kubernetes backend", func() {
...
+			Delete().
+			Resource("networkpolicies").
+			Namespace("default").
+			Name(np.ObjectMeta.Name).
```
Comment (gunjan5) [inline lib/backend/k8s/k8s_test.go]: why don't we have `Timeout(10 * time.Second).` here?

File: lib/backend/k8s/k8s_test.go [hunk 4]
```diff
@@ -589,13 +578,6 @@ var _ = Describe("Test Syncer API for Kubernetes backend", func() {
 			},
 		}
 
-		// Make sure we clean up after ourselves.  We allow this to fail because
-		// part of our explicit testing below is to delete the resource.
-		defer func() {
-			gnpClient.Delete(ctx, kvp1a.Key, "")
```
Comment (gunjan5) [inline lib/backend/k8s/k8s_test.go]: where is the equivalent code for this (in `AfterEach`?)?

Comment (caseydavenport) [reply lib/backend/k8s/k8s_test.go]: This is part of the `c.Clean` call which cleans up all Calico resources.

File: lib/backend/k8s/k8s_test.go (hunk 2 above)
Comment (caseydavenport) [reply lib/backend/k8s/k8s_test.go]: Yea, should just remove this actually.

File: lib/backend/k8s/k8s_test.go (hunk 4 above)
Comment (gunjan5) [reply lib/backend/k8s/k8s_test.go]: ah yes! right right

File: lib/backend/k8s/k8s_test.go (hunk 1 above)
Comment (caseydavenport) [reply lib/backend/k8s/k8s_test.go]: I suppose we could, but that's extra work for questionable gain. I think we should just clean up the things we know we may have created for now.

**Comment by caseydavenport** [comment]
@gunjan5 all cleaned up

**Pull request merged**